In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [3]:
players<- read_csv("players.csv")

sessions<- read_csv("sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


The dataset contains **196 observations**, with each representing an individual who participated in an study involving an online Minecraft server. There is a total of **7 variables**.

### List of Variables

| Variable        | Description |
|----------------|-------------|
| `experience`    | Self-reported skill level in Minecraft. It is either: Beginner, Amateur, Regular, Pro, or Veteran. |
| `subscribe`     | Whether the given participant subscribed to a gaming-related newsletter. Either: Yes or No. |
| `hashedEmail`   | An encrypted version of a participants email (hashed). Each email is unique. |
| `played_hours`  | The total number of hours of Minecraft played during data collection. |
| `name`          | The participant’s name. |
| `gender`        | The participant’s gender. |
| `Age`           | The participant’s age in years. |

| Variable              | Description |
|-----------------------|-------------|
| `hashedEmail`         | An encrypted version of a participants email (hashed). Each email is unique. |
| `start_time`          | The session start time, formatted as `dd/mm/yyyy hh:mm`. |
| `end_time`            | The session end time, formatted as `dd/mm/yyyy hh:mm`. |
| `original_start_time` | The raw numeric timestamp representing the start time (POSIX style format). |
| `original_end_time`   | The raw numeric timestamp representing the end time. |

### Data Collection Process

The data was collected from a free public Minecraft server operated from researchers at the University of British Columbia. The goal of this data collection was to gather behavioral and demographic information of players to better support the development of an AI capable of playing Minecraft.

### Data Quality

Overall, the dataset is high quality and full of insightful information. However, there are some issues with the naming of variables:
- Some variables use `snake_case` (e.g., `played_hours`), others use `camelCase` (e.g., `hashedEmail`), and some are inconsistently capitalized (e.g., `Age`).
- Overall, these issues are minor and do not actually affect the usability of the dataset.

"Which Has a Greater Effect on a Player Subscribing to a Game Newsletter: Hours Played or Total Experience?"


**Predicting Whether A Player Is Likely To Be Subscribed**

In today’s digital gaming world, understanding how users behave is key to boosting engagement and keeping players around. Many gaming platforms use subscription-based models to make money, so knowing what motivates a user to subscribe can guide effective strategies. In this project, we look at how a player’s experience level and their activity during sessions can affect their chances of subscribing. By carefully processing data, visualizing it, and using predictive modeling with k-Nearest Neighbors (KNN), we aim to find important behavior patterns and create a model to predict subscription status.

### Question

Can a player’s experience level and session activity predict whether they are a subscriber or not?

We examine this by modeling the relationship between the **response variable**:

- `subscribe` (Yes/No)

and the **explanatory variables**:

- `experience` (categorical)
- `total_session_time_hrs` (numeric)
- `average_session_time_hrs` (numeric)
- `num_of_sessions` (numeric) 